### PR TITLE
Enable parallel test

### PR DESCRIPTION
**Pull request by jihoonson**
Here is the comparison of testing time.

*Sequential test*
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] io.druid:druid ..................................... SUCCESS [  4.468 s]
[INFO] java-util .......................................... SUCCESS [ 12.232 s]
[INFO] druid-api .......................................... SUCCESS [  4.750 s]
[INFO] druid-common ....................................... SUCCESS [  8.116 s]
[INFO] bytebuffer-collections ............................. SUCCESS [  3.654 s]
[INFO] druid-processing ................................... SUCCESS [05:36 min]
[INFO] druid-aws-common ................................... SUCCESS [  0.478 s]
[INFO] druid-server ....................................... SUCCESS [02:44 min]
[INFO] druid-examples ..................................... SUCCESS [  0.488 s]
[INFO] druid-indexing-hadoop .............................. SUCCESS [01:19 min]
[INFO] druid-indexing-service ............................. SUCCESS [01:40 min]
[INFO] druid-services ..................................... SUCCESS [  7.252 s]
[INFO] druid-s3-extensions ................................ SUCCESS [  3.767 s]
[INFO] druid-datasketches ................................. SUCCESS [03:12 min]
[INFO] druid-kafka-eight .................................. SUCCESS [  0.168 s]
[INFO] druid-histogram .................................... SUCCESS [  7.000 s]
[INFO] mysql-metadata-storage ............................. SUCCESS [  0.417 s]
[INFO] druid-kafka-indexing-service ....................... SUCCESS [07:31 min]
[INFO] druid-integration-tests ............................ SUCCESS [  1.922 s]
[INFO] druid-benchmarks ................................... SUCCESS [  5.773 s]
[INFO] druid-avro-extensions .............................. SUCCESS [  7.252 s]
[INFO] druid-hdfs-storage ................................. SUCCESS [  8.468 s]
[INFO] druid-stats ........................................ SUCCESS [  6.449 s]
[INFO] druid-lookups-cached-global ........................ SUCCESS [ 46.248 s]
[INFO] druid-kafka-extraction-namespace ................... SUCCESS [  8.839 s]
[INFO] postgresql-metadata-storage ........................ SUCCESS [  1.423 s]
[INFO] druid-lookups-cached-single ........................ SUCCESS [ 11.649 s]
[INFO] druid-azure-extensions ............................. SUCCESS [  2.771 s]
[INFO] druid-cassandra-storage ............................ SUCCESS [  0.312 s]
[INFO] druid-rocketmq ..................................... SUCCESS [  0.317 s]
[INFO] druid-cloudfiles-extensions ........................ SUCCESS [  2.487 s]
[INFO] graphite-emitter ................................... SUCCESS [  2.338 s]
[INFO] druid-kafka-eight-simple-consumer .................. SUCCESS [  0.199 s]
[INFO] druid-rabbitmq ..................................... SUCCESS [  3.102 s]
[INFO] druid-distinctcount ................................ SUCCESS [  2.392 s]
[INFO] druid-parquet-extensions ........................... SUCCESS [  6.997 s]
[INFO] statsd-emitter ..................................... SUCCESS [  2.025 s]
[INFO] druid-orc-extensions ............................... SUCCESS [ 12.209 s]
[INFO] druid-time-min-max ................................. SUCCESS [  8.253 s]
[INFO] druid-google-extensions ............................ SUCCESS [  2.269 s]
[INFO] druid-virtual-columns .............................. SUCCESS [  2.522 s]
[INFO] druid-caffeine-cache ............................... SUCCESS [  5.903 s]
[INFO] distribution ....................................... SUCCESS [  0.126 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 25:32 min
[INFO] Finished at: 2016-12-13T09:39:05+09:00
[INFO] Final Memory: 213M/1506M
[INFO] ------------------------------------------------------------------------
```

*Parallel test*
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] io.druid:druid ..................................... SUCCESS [  0.959 s]
[INFO] java-util .......................................... SUCCESS [  4.915 s]
[INFO] druid-api .......................................... SUCCESS [  2.312 s]
[INFO] druid-common ....................................... SUCCESS [  4.908 s]
[INFO] bytebuffer-collections ............................. SUCCESS [  1.739 s]
[INFO] druid-processing ................................... SUCCESS [02:33 min]
[INFO] druid-aws-common ................................... SUCCESS [  0.170 s]
[INFO] druid-server ....................................... SUCCESS [01:15 min]
[INFO] druid-examples ..................................... SUCCESS [  0.145 s]
[INFO] druid-indexing-hadoop .............................. SUCCESS [ 46.378 s]
[INFO] druid-indexing-service ............................. SUCCESS [ 51.878 s]
[INFO] druid-services ..................................... SUCCESS [  4.272 s]
[INFO] druid-s3-extensions ................................ SUCCESS [  2.834 s]
[INFO] druid-datasketches ................................. SUCCESS [01:36 min]
[INFO] druid-kafka-eight .................................. SUCCESS [  0.072 s]
[INFO] druid-histogram .................................... SUCCESS [  5.494 s]
[INFO] mysql-metadata-storage ............................. SUCCESS [  0.136 s]
[INFO] druid-kafka-indexing-service ....................... SUCCESS [03:33 min]
[INFO] druid-integration-tests ............................ SUCCESS [  3.199 s]
[INFO] druid-benchmarks ................................... SUCCESS [  2.160 s]
[INFO] druid-avro-extensions .............................. SUCCESS [  4.334 s]
[INFO] druid-hdfs-storage ................................. SUCCESS [  5.311 s]
[INFO] druid-stats ........................................ SUCCESS [  3.624 s]
[INFO] druid-lookups-cached-global ........................ SUCCESS [ 29.681 s]
[INFO] druid-kafka-extraction-namespace ................... SUCCESS [  6.016 s]
[INFO] postgresql-metadata-storage ........................ SUCCESS [  0.882 s]
[INFO] druid-lookups-cached-single ........................ SUCCESS [  6.526 s]
[INFO] druid-azure-extensions ............................. SUCCESS [  1.372 s]
[INFO] druid-cassandra-storage ............................ SUCCESS [  0.085 s]
[INFO] druid-rocketmq ..................................... SUCCESS [  0.064 s]
[INFO] druid-cloudfiles-extensions ........................ SUCCESS [  1.375 s]
[INFO] graphite-emitter ................................... SUCCESS [  1.318 s]
[INFO] druid-kafka-eight-simple-consumer .................. SUCCESS [  0.061 s]
[INFO] druid-rabbitmq ..................................... SUCCESS [  0.977 s]
[INFO] druid-distinctcount ................................ SUCCESS [  1.292 s]
[INFO] druid-parquet-extensions ........................... SUCCESS [  3.193 s]
[INFO] statsd-emitter ..................................... SUCCESS [  0.981 s]
[INFO] druid-orc-extensions ............................... SUCCESS [  6.955 s]
[INFO] druid-time-min-max ................................. SUCCESS [  3.774 s]
[INFO] druid-google-extensions ............................ SUCCESS [  1.357 s]
[INFO] druid-virtual-columns .............................. SUCCESS [  1.421 s]
[INFO] druid-caffeine-cache ............................... SUCCESS [  2.928 s]
[INFO] distribution ....................................... SUCCESS [  0.070 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 12:34 min
[INFO] Finished at: 2016-12-14T10:08:09+09:00
[INFO] Final Memory: 89M/1536M
[INFO] ------------------------------------------------------------------------
```

---
This change is [Reviewable](https://reviewable.io/reviews/druid-io/druid/3774)

### CODE REVIEW

**Comment by fjy** [comment]
👍

**Comment by jihoonson** [comment]
I set the number of JVM processes as 2 by default. It can be changed by setting ```maven.fork.count```.

**Comment by jon-wei** [comment]
@jihoonson there's travis failures

**Comment by jihoonson** [comment]
Yes, it is a simple test failure in SocketUtilTest. I'm fixing it.

**Comment by jihoonson** [comment]
@jon-wei thanks. fixed.

File: common/src/main/java/io/druid/common/utils/SocketUtil.java [hunk 1]
```diff
@@ -21,15 +21,24 @@
...
  */
 public class SocketUtil
 {
-  public static int findOpenPort(int startPort)
+
+  private static final Random rnd = new Random(System.currentTimeMillis());
+
+  public static int findOpenPort(int basePort)
   {
+    final int startPort = basePort == -1 ? -1 : rnd.nextInt(0x7fff) + basePort;
```
Comment (jon-wei) [inline common/src/main/java/io/druid/common/utils/SocketUtil.java]: let's make this check for all negatives instead of just -1

Comment (jihoonson) [inline common/src/main/java/io/druid/common/utils/SocketUtil.java]: Fixed. Thanks!

**Comment by jon-wei** [comment]
:+1:

**Comment by leventov** [comment]
Parallel tests didn't actually run on Travis, because of https://github.com/travis-ci/travis-ci/issues/1400#issuecomment-251242624

**Comment by jihoonson** [comment]
@leventov that issue seems to relate to mis-overriding ```install``` step. Our travis looks working well with the parallelizing option. [Here](https://travis-ci.org/druid-io/druid/builds/230140537#L3852) is an example. You can see [two processes for testing different classes start simultaneously](https://travis-ci.org/druid-io/druid/builds/230140537#L3961).

**Pull request merged**